armi/reactor/components/__init__.py

```diff
@@ -186,6 +187,23 @@ def getBoundingCircleOuterDiameter(self, Tc=None, cold=False):
         """
         return 2 * math.sqrt(self.getComponentArea(cold=cold) / math.pi)
 
+    def getCircleInnerDiameter(self, Tc=None, cold=False):
+        """
+        Approximate the component as circular; i.e., inner diameter is zero.
+
+        Parameters
+        ----------
+        Tc : float
+            Ignored for this component
+        cold : bool, optional
+            If True, compute the area with as-input dimensions, instead of thermally-expanded.
+
+        Notes
+        -----
+        Tc is not used in this method for this particular component.
+        """
+        return 0.0
```

Comment:
This seems like an okay thing for a shaped component to have to define on their own. Maybe a `DerivedShape` is the only exception.

Saying this because we `try`/`catch` a `NotImplementedError` below so if  `Component.getCircleInnerDiameter` is required for components, we'd be okay in that context.

But that could be a larger change that is likely better served outside this PR.

Non-blocking comment. If there's traction I can make a ticket

Reply:
The assumption for the unshaped component is to return the smallest possible bounding circle outer diameter, i.e., assume the component has a disk shape. It seems consistent with that assumption to assume that the inner diameter is 0.0.

The exception is caught in `__lt__`, but only for the purpose of producing a clearer error message. The error is still raised and the code crashes. So we still need to be able to sort `UnshapedComponent`s.